ext/civi_contribute/Civi/Api4/Service/ContributionTasksProvider.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+namespace Civi\Api4\Service;
+
+use CRM_Contribute_ExtensionUtil as E;
+use Civi\Core\Event\GenericHookEvent;
+
+/**
+ * @service
+ * @internal
+ */
+class ContributionTasksProvider extends \Civi\Core\Service\AutoSubscriber {
+
+  public static function getSubscribedEvents(): array {
+    return [
+      'hook_civicrm_searchKitTasks' => 'addContributionTasks',
+    ];
+  }
+
+  /**
+   * @param \Civi\Core\Event\GenericHookEvent $event
+   */
+  public function addContributionTasks(GenericHookEvent $event): void {
+    // FIXME: CRM_Contribute_Task::tasks() should respect `$this->checkPermissions`
+    foreach (\CRM_Contribute_Task::tasks() as $id => $task) {
```

Comment:
@colemanw This approach can't be generalized because `CRM_Contribute_Task()` a) extends `CRM_Core_Task` and b) the result is cached in `CRM_Core_Task`. I tried applying this approach to Grants and there was cache poisoning.

In #30321 I change the scope of the cache in `CRM_Contribute_Task()`. Without this, different `hook_civicrm_searchKitTasks` implementation using `CRM_X_Task::tasks()` will clobber each other.